app/views/wikis/edit.html.haml

```diff
@@ -9,5 +9,13 @@
       .form-group
         = f.label :body
         = f.text_area :body, class: "form-control"
+
+      - if current_user.role == 'admin' || current_user.role == 'premium'
+        .checkbox
+          = f.select :private, [[false, :public], [true, :private]]
```

Comment:
Assuming we want the text for the 2 options to be "Public" and "Private", this I think would be closer to what's needed:

```haml
  = f.select :private, [["Public", false], ["Private", true]]
``
```

The rules for defining that array of options `[["Public", false], ["Private", true]]` are in the (`options_for_select`)[http://api.rubyonrails.org/classes/ActionView/Helpers/FormOptionsHelper.html#method-i-options_for_select] docs.